src/octoprint/util/gcodeInterpreter.py

```diff
@@ -120,6 +122,13 @@ def _load(self, gcodeFile, printer_profile, throttle=None):
 					z = getCodeFloat(line, 'Z')
 					e = getCodeFloat(line, 'E')
 					f = getCodeFloat(line, 'F')
+					#Check if x,y or z exceeds printing area
+					width=float(printer_profile["volume"]["width"])
+					depth=float(printer_profile["volume"]["depth"])
+					height=float(printer_profile["volume"]["height"])
+					if not self.warn and (x>width or y>depth or z>height):
```

Comment:
This only works for print beds which have their `(0,0)` origin in the lower left corner of the print bed. This is not necessarily the case, it could also be the center of the print bed, in which case a simple check if x/y/z are larger than width/depth/height would produce a false negative (imagine a printbed of 20x20 cm, center in the middle, so 10cm up and left from the lower left corner - an object with max X coordinate of 19 would definitely not fit since that would be 19cm to the right of the origin, so actually 9cm off the bed, but the code would not detect that).